main.js

```diff
@@ -1,82 +1,190 @@
+function removeChoice($choicesList, $choice) {
+  $choice.setAttribute('class', 'choice card-panel animated fadeOut')
+  setTimeout(function () {
+    $choicesList.removeChild($choice)
+  }, 1000)
+}
 
-    $unorderedList.appendChild($list)
-    $list.appendChild($removeButton)
+function randomNumber(data) {
+  if (data.length !== 0) {
+    var number = Math.round(Math.random() * (data.length - 1))
+  }
+  return number
+}
 
-    $removeButton.addEventListener('click', function (e) {
-      $unorderedList.removeChild($list)
-    })
+function renderDecision(data, number) {
+  for (var i = 0; i < data.length; i++) {
+    if (data[i].id === number) {
+      var decision = {
+        winningId: data[i].id,
+        winningChoice: data[i].userChoice,
+        winner: data[i].userName,
+        choices: data
+      }
+      return decision
+    }
   }
 }
 
-function clearList($unorderedList, $lists) {
-  for (var i = 0; i < $lists.length; i++) {
-    $unorderedList.removeChild($lists[i])
+function fadeOutOtherChoices(data) {
+  var $choices = document.querySelectorAll('.choice')
+  for (var i = 0; i < $choices.length; i++) {
+    var choiceDataId = $choices[i].getAttribute('data-id')
+    if (choiceDataId !== data.winningId.toString()) {
+      $choices[i].setAttribute('class', 'choice card-panel animated fadeOut')
+    }
+    else {
+      $choices[i].setAttribute('class', 'choice card-panel')
+    }
   }
 }
 
-function randomizeList($unorderedList, $lists) {
-  var index = Math.round(Math.random() * ($lists.length - 1))
+function disableButton($button) {
+  if ($button.getAttribute('id') === 'remove') {
+    $button.setAttribute('class', 'btn-floating btn-large waves-effect waves-light red disabled')
+  }
+  else {
+    $button.setAttribute('class', 'waves-effect waves-light btn-large disabled')
+  }
+}
 
-  for (var i = 0; i < $lists.length; i++) {
-    if (index !== i) {
-      $unorderedList.removeChild($lists[i])
-    }
+function enableButton($button) {
+  if ($button.getAttribute('id') === 'remove') {
+    $button.setAttribute('class', 'btn-floating btn-large waves-effect waves-light red')
   }
+  else {
+    $button.setAttribute('class', 'waves-effect waves-light btn-large')
+  }
+}
+
+function clearUserInput($userChoiceInput, $userNameInput) {
+  $userChoiceInput.value = ''
+  $userNameInput.value = ''
 }
 
+var $submitButton = document.querySelector('#submit')
 $submitButton.addEventListener('click', function (e) {
-  var $choiceList = document.querySelector('.choice-list')
-  var $choice = document.createElement('li')
-  $choice.setAttribute('class', 'choice col s12 animated bounceInUp')
-  renderList($choiceList, $choice)
+  e.preventDefault()
+  var $userChoiceInput = document.querySelector('#user-choice')
+  var $userNameInput = document.querySelector('#user-name')
+  var choiceInfo = createChoiceInfo($userChoiceInput, $userNameInput)
+  if (choiceInfo) {
+    choices.push(choiceInfo)
+    var $choice = choice(choiceInfo)
```

Comment:
Why can't the code that makes the choice also call the removeButton function and add the button?